cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/analyzer/ReplicationFactorChangeTest.java

```diff
@@ -168,22 +165,19 @@ public ReplicationFactorChangeTest(int testId,
   }
 
   private static boolean expectedToOptimize(short replicationFactor, Class<? extends Goal> goalClass, boolean smallCluster) {
-    if ((replicationFactor == SMALL_REPLICATION_FACTOR && goalClass == ReplicaDistributionGoal.class && smallCluster)
-        || (replicationFactor == SMALL_REPLICATION_FACTOR && goalClass == DiskUsageDistributionGoal.class)
-        || (replicationFactor == SMALL_REPLICATION_FACTOR && goalClass == NetworkInboundUsageDistributionGoal.class)
-        || (replicationFactor == SMALL_REPLICATION_FACTOR && goalClass == NetworkOutboundUsageDistributionGoal.class)
-        || (replicationFactor == SMALL_REPLICATION_FACTOR && goalClass == CpuUsageDistributionGoal.class)
-        || (replicationFactor == SMALL_REPLICATION_FACTOR && goalClass == LeaderReplicaDistributionGoal.class && smallCluster)
-        || (replicationFactor == LARGE_REPLICATION_FACTOR && goalClass == NetworkOutboundUsageDistributionGoal.class && smallCluster)
-        || (goalClass == LeaderBytesInDistributionGoal.class && (replicationFactor == SMALL_REPLICATION_FACTOR || smallCluster))
-        || (replicationFactor == LARGE_REPLICATION_FACTOR && goalClass == DiskUsageDistributionGoal.class && !smallCluster)
-        || (replicationFactor == LARGE_REPLICATION_FACTOR && goalClass == NetworkInboundUsageDistributionGoal.class && !smallCluster)
-        || (replicationFactor == LARGE_REPLICATION_FACTOR && goalClass == CpuUsageDistributionGoal.class)
-        || (replicationFactor == SMALL_REPLICATION_FACTOR && goalClass == MinTopicLeadersPerBrokerGoal.class)
-        || (replicationFactor == LARGE_REPLICATION_FACTOR && goalClass == MinTopicLeadersPerBrokerGoal.class && !smallCluster)) {
-      return false;
-    }
-    return true;
+    return (replicationFactor != SMALL_REPLICATION_FACTOR || goalClass != ReplicaDistributionGoal.class || !smallCluster) && (
+        replicationFactor != SMALL_REPLICATION_FACTOR || goalClass != DiskUsageDistributionGoal.class) && (
```

Comment:
```suggestion
    // Return true by default. Return false when criteria are met in any of the following rows.
    return (replicationFactor != SMALL_REPLICATION_FACTOR || goalClass != ReplicaDistributionGoal.class || !smallCluster) 
           && (replicationFactor != SMALL_REPLICATION_FACTOR || goalClass != DiskUsageDistributionGoal.class) && (
```

Reply:
Addressed the feedback regarding moving the `&&` to the beginning of the line. But omitted the comment as it was not clear to me -- i.e. if any line is false, wouldn't the function returns false?

Reply:
Yes. But I think the original code was easier to read. I can tell at a glance that the default is to return true. 
In the new code I have to carefully see through and puzzle out all the booleans operator to make sure that is the default. 
Adding the comment improves the readability to me.
Maybe we can just omit the 2nd sentence and just add this? `// Return true by default.`

Reply:
Thanks for the feedback!
Added a JavaDoc to explain how this method is expected to behave under (1) typical cases and (2) exceptional cases. Also changed the `&&`s back to `||`s since the logic seems more clear that way.